lib/bpl/transformation/shadowing.rb

```diff
@@ -26,6 +26,8 @@ def shadow_copy(node)
         shadow
       end
 
+      def shadow_expr_eq(expr) "#{expr} == #{shadow_copy(expr)}" end
```

Comment:
Done with option 1 (use "bpl" to turn strings into nodes). I can go back on
that decision if you prefer, but not tonight.

On Fri, 18 Sep 2015 at 20:27 Michael Emmi <notifications@github.com> wrote:

> In lib/bpl/transformation/shadowing.rb
> <https://github.com/michael-emmi/bam-bam-boogieman/pull/3#discussion_r39885805>
> :
>
> > @@ -26,6 +26,8 @@ def shadow_copy(node)
> >          shadow
> >        end
> >
> > +      def shadow_expr_eq(expr) "#{expr} == #{shadow_copy(expr)}" end
>
> Ah, haha, good point!
>
> Two options then:
>
>    1.
>
>    convert those strings to Nodes first via the bpl method — fixing the
>    argument type of shadow_eq to be Node.
>    2.
>
>    allow the argument type to be Node or string, and handle both cases in
>    shadow_copy.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/michael-emmi/bam-bam-boogieman/pull/3/files#r39885805>
> .
>